rio/src/RIO/Prelude/Extra.hs

```diff
@@ -75,15 +81,28 @@ nubOrd =
 
 -- | Run the second value if the first value returns 'True'
 whenM :: Monad m => m Bool -> m () -> m ()
-whenM boolM action = do
-  x <- boolM
-  if x then action else return ()
+whenM boolM action = boolM >>= (`when` action)
 
 -- | Run the second value if the first value returns 'False'
 unlessM :: Monad m => m Bool -> m () -> m ()
-unlessM boolM action = do
-  x <- boolM
-  if x then return () else action
+unlessM boolM action = boolM >>= (`unless` action)
+
+
+-- | Same as `(>>)`, but with arguments flipped.
+--
+-- @since 0.1.10.0
+(<<) :: Monad m => m a -> m b -> m a
+(<<) = flip (>>)
+
+infixr 1 <<
+
+#if !MIN_VERSION_base(4, 11, 0)
```

Comment:
This is a nice idea. However, it can cause naming conflicts with other libraries that do the same thing, e.g. lens. Maybe that's not so important. I've definitely wanted this operator before...

Reply:
lol ok